### ISSUE TITLE
Implement support for VS 2015 and ASP.NET vNext

### ISSUE THREAD

**Issue by pauliusmaciulis** [devbridge/BetterCMS#1277]

**Comment by clmcgrath** [comment]
will this be a top to bottom redesign ?
there a lot of features that coiuld be taken advantage of with next such as adding support for DI and dropping webforms authentication, which i believe is being deprecated in asp.net 5/vnext projects

**Comment by pauliusmaciulis** [comment]
Yes, it will be full review of the source base and a top to bottom redesign. The API layer will be the same or will have only minimal changes, the modular principles will be the same, but overall totally - v2.0 should take advantages of the vNext.

**Comment by gerbenis** [comment]
BetterCMS migration to vNext (feature/1277_vNext)

What has been done:
- [BetterModules](https://github.com/devbridge/BetterModules) early alpha version (core functionality is  working);
- Moved BetterCms.Configuration project to support ASP.NET 5 Options pattern;
- Moved BetterCms.Core (couple of places are still waiting for review, but major functionality has been moved);
- Moved MvcContrib Grids to BetterCms.Core for MVC6 support (MvcContrib has their own feature branch for MVC6 support, maybe they will have finished it by the time we come back to migrating Cms to vNext);
- Started moving Cms Root Module.

What needs to be done:
- Implement View Precompilation in [BetterModules](https://github.com/devbridge/BetterModules);
- Implement Caching (I suggest to use ASP.NET 5 Caching and extend its behavior with extension methods or sth like that);
- Review [BetterModules](https://github.com/devbridge/BetterModules) project for redundant pieces of code (some methods and services, such as IHttpContextAccessor, seem to be unused);
- Move Authentication and Authorization (FormsAuthentication is gone, now we have CookieAuthentication middleware instead. But maybe want to use sth else?);
- Rethink how Projections are registered and resolved from DI container (ASP.NET 5 DI are not planning to implement named services);
- Finish moving Root Module;
- Finish moving Other modules;
- Implement Cms services registration (AddBetterCms extension method);
- Implement Cms registration to the pipeline (UseBetterCms extension method).

Most of issues I have faced so far are written as TODOs in code.

**Comment by Meligy** [comment]
> Implement Caching (I suggest to use ASP.NET 5 Caching and extend its behavior with extension methods or sth like that);

Does this mean support is considered for full framework only? As in, no CoreCLR?

(Not that I'm requesting it or anything, just making this bit visible)

**Comment by smikulenas** [comment]
Caching is in CoreCLR, so not a big deal. Any way, currently the only obstacle that we see regarding CoreCLR vs full framework - NHibernate, as currently it does not support CoreCLR.

**Comment by brightertools** [comment]
Hi, What us the status on porting to .netcore?

I personally am not too bothered with dependencies on the full framework, it would be interesting to hear people's views on this?

Have you considered using EF 7? 
I appreciate this may be a big change but,I have done fair it of work to create a framework using T4 templates to generate core CRUD, with features. Would be happy to share this, but would need to talk this through with someone responsible for data access.

Using T4 means Visual Studio only for dev though, which again is probably not a bad thing.
